testing/ec2_mocks.go

```diff
@@ -64,3 +64,25 @@ func (m *MockEC2Client) DescribeInstances(input *ec2.DescribeInstancesInput) (ou
 
 	return output, nil
 }
+
+func (m *MockEC2Client) DescribeInstancesPages(input *ec2.DescribeInstancesInput, fn func(*ec2.DescribeInstancesOutput, bool) bool) error {
+	var err error = nil
+	var continueIterating bool = true
+
+	// Grab initial case and start looping
+	output, err := m.DescribeInstances(input)
+	for err == nil && continueIterating {
+		continueIterating = fn(output, (output.NextToken == nil))
+
+		// Just keep chugging unless we are at the end of the page.
+		if output.NextToken == nil || !continueIterating {
```

Comment:
This conditional does not need an else. The code there can be treated as the "happy path" since the if will break out of the loop if true